runtime/roots_nat.c

```diff
@@ -79,14 +80,19 @@ static link* frametables_list_tail(link *list) {
 }
 
 static frame_descr * next_frame_descr(frame_descr * d) {
-  uintnat nextd;
-  nextd =
-    ((uintnat)d +
-     sizeof(char *) + sizeof(short) + sizeof(short) +
-     sizeof(short) * d->num_live + sizeof(frame_descr *) - 1)
-    & -sizeof(frame_descr *);
-  if (d->frame_size & 1) nextd += sizeof(void *); /* pointer to debuginfo */
-  return((frame_descr *) nextd);
+  unsigned char num_allocs = 0, *p;
+  CAMLassert(d->retaddr >= 4096);
```

Comment:
It is indeed true because of the size of pages. During development, this caught several bugs that the `!= 0` check missed, where a misalignment of frametable data caused the retaddr to contain a small integer (e.g. frame size) instead of the return address.